openvino_xai/metrics/adcc.py

```diff
@@ -0,0 +1,132 @@
+        saliency_map_mapped_image = saliency_map_mapped_image.saliency_map[class_idx]
+
+        A, B = saliency_map, saliency_map_mapped_image
+        # Pearson correlation coefficient
+        Asq, Bsq = A.flatten(), B.flatten()
+        y, _ = STS.pearsonr(Asq, Bsq)
+        y = (y + 1) / 2
+
+        return y
+
+    @staticmethod
+    def complexity(saliency_map: np.ndarray) -> float:
+        """
+        Measures the complexity of the saliency map. Less important pixels -> less complexity.
+        Defined as L1 norm of the saliency map.
+        The less the better.
+        """
+        return abs(saliency_map).sum() / (saliency_map.shape[-1] * saliency_map.shape[-2])
+
+    def __call__(self, saliency_map: np.ndarray, class_idx: int, input_image: np.ndarray) -> Dict[str, float]:
+        """
+        Calculate the ADCC metric for a given saliency map and class index.
+        The more the better.
+
+        Parameters:
+        :param saliency_map: Saliency map for class_idx class (H, W).
+        :type saliency_map: np.ndarray
+        :param class_idx: The class index of saliency map.
+        :type class_idx: int
+        :param input_image: The input image to the model (H, W, C).
+        :type input_image: np.ndarray
+
+        Returns:
+        :return: A dictionary containing the ADCC, coherency, complexity, and average drop metrics.
+        :rtype: Dict[str, float]
+        """
+        if not (0 <= np.min(saliency_map) and np.max(saliency_map) <= 1):
+            # Scale saliency map to [0, 1]
+            saliency_map = saliency_map / 255
+
+        model_output = self.model_predict(input_image)
+
+        avgdrop = self.average_drop(saliency_map, class_idx, input_image, model_output)
+        coh = self.coherency(saliency_map, class_idx, input_image)
+        com = self.complexity(saliency_map)
+
+        adcc = 3 / (1 / coh + 1 / (1 - com) + 1 / (1 - avgdrop))
+        return {"adcc": adcc, "coherency": coh, "complexity": com, "average_drop": avgdrop}
+
+    def evaluate(
+        self, explanations: List[Explanation], input_images: List[np.ndarray], **kwargs: Any
+    ) -> Dict[str, float]:
+        """
+        Evaluate the ADCC metric over a list of explanations and input images.
+
+        Parameters:
+        :param explanations: A list of explanations for each image.
+        :type explanations: List[Explanation]
+        :param input_images: A list of input images.
+        :type input_images: List[np.ndarray]
+
+        Returns:
+        :return: A dictionary containing the average ADCC score.
+        :rtype: Dict[str, float]
+        """
+        results = []
+        for input_image, explanation in zip(input_images, explanations):
+            for class_idx, saliency_map in explanation.saliency_map.items():
+                metric_dict = self(saliency_map, int(class_idx), input_image)
+                results.append(metric_dict["adcc"])
+        adcc = np.mean(np.array(results), axis=0)
```

Comment:
I'm not sure which is right, but we could alternatively compute ADCC based on the averages of sub metrics.
Simply put, average of ADCC vs ADCC of average of sub metrics.
Mark implemented like the latter approach for the dataset evaluation.
There is no reference for multi-image evaluation in the original repo, unfortunately.

Reply:
Thank you for this point. The metric should be the same in both cases and the performance difference should not be visible. 

I personally think that it's better to average the adcc so `__call__` method will return `adcc` for one saliency map as well. I don't know why (and if) we need that, but I tried to keep the similarity between` __call__` and `evaluate` methods